wordcount/wordcount.go

```diff
@@ -1,62 +1,125 @@
 package main
 
 import (
+	"log"
 	"github.com/pauloaguiar/ces27-lab1/mapreduce"
 	"hash/fnv"
+	"regexp"
+	"strings"
+	"strconv"
 )
 
-// mapFunc is called for each array of bytes read from the splitted files. For wordcount
-// it should convert it into an array and parses it into an array of KeyValue that have
-// all the words in the input.
+/**
+ * mapFunc is called for each array of bytes read from the splitted files. For wordcount
+ * it should convert it into an array and parses it into an array of KeyValue that have
+ * all the words in the input.
+ */
 func mapFunc(input []byte) (result []mapreduce.KeyValue) {
-	// 	Pay attention! We are getting an array of bytes. Cast it to string.
-	//
-	// 	To decide if a character is a delimiter of a word, use the following check:
-	//		!unicode.IsLetter(c) && !unicode.IsNumber(c)
-	//
-	//	Map should also make words lower cased:
-	//		strings.ToLower(string)
-	//
-	// IMPORTANT! The cast 'string(5)' won't return the character '5'.
-	// 		If you want to convert to and from string types, use the package 'strconv':
-	// 			strconv.Itoa(5) // = "5"
-	//			strconv.Atoi("5") // = 5
+	/*
+	 * Pay attention! We are getting an array of bytes. Cast it to string.
+	 *
+	 * To decide if a character is a delimiter of a word, use the following check:
+	 * !unicode.IsLetter(c) && !unicode.IsNumber(c)
+	 *
+	 * Map should also make words lower cased:
+	 * 	strings.ToLower(string)
+	 *
+	 * IMPORTANT! The cast 'string(5)' won't return the character '5'.
+	 *		If you want to convert to and from string types, use the package 'strconv':
+	 *			strconv.Itoa(5) // = "5"
+	 *			strconv.Atoi("5") // = 5
+	 */
 
-	/////////////////////////
-	// YOUR CODE GOES HERE //
-	/////////////////////////
+	/*
+	 * Casting byte array to string
+	 */
+	s := string(input[:])
+
+	/*
+	 * Clearing text using regular expression
+	 * Removing all invalid characters
+	 * Replacing set of white space by one white space 
+	 */
+	for _,rexp := range []string{"([^A-Za-z]+)", " +"} {
+		
+		reg, err := regexp.Compile(rexp)
+
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		s = reg.ReplaceAllString(s, " ")
+	}
```

Comment:
This code is not giving the same result as using
```go
!unicode.IsLetter(c) && !unicode.IsNumber(c)
```
to find word delimiters. I would say that getting it to work exactly the same using regex is going to be hard as the unicode.IsLetter function can be pretty complex to write as a simple regex query.